arches/app/models/resource.py

```diff
@@ -189,22 +189,31 @@ def bulk_save(resources):
             resource.tiles = resource.get_flattened_tiles()
             tiles.extend(resource.tiles)
 
-        # need to save the models first before getting the documents for index
+        # need to handle if the bulk load is appending tiles to existing resources/
+        existing_resources = Resource.objects.filter(resourceinstanceid__in=[resource.resourceinstanceid for resource in resources])
+        existing_resources_ids = [existing_resource.resourceinstanceid for existing_resource in existing_resources]
+        resources_to_create = [resource for resource in resources if resource.resourceinstanceid not in existing_resources_ids]
+
         start = time()
-        Resource.objects.bulk_create(resources)
+        Resource.objects.bulk_create(resources_to_create)
         TileModel.objects.bulk_create(tiles)
 
-        print(f"Time to bulk create tiles and resources: {datetime.timedelta(seconds=time() - start)}")
+        logger.info(f"Time to bulk save tiles and resources: {datetime.timedelta(seconds=time() - start)}")
 
         start = time()
-        for resource in resources:
+        for resource in resources_to_create:
             resource.save_edit(edit_type="create")
 
-        resources[0].tiles[0].save_edit(note=f"Bulk created: {len(tiles)} for {len(resources)} resources.", edit_type="bulk_create")
+        try:
+            resources[0].tiles[0].save_edit(note=f"Bulk created: {len(tiles)} for {len(resources)} resources.", edit_type="bulk_create")
+        except:
+            pass
 
-        print("Time to save resource edits: %s" % datetime.timedelta(seconds=time() - start))
+        logger.info("Time to save resource edits: %s" % datetime.timedelta(seconds=time() - start))
+
+        # refetch new AND updated resources and index
+        for resource in Resource.objects.filter(resourceinstanceid__in=[resource.resourceinstanceid for resource in resources]):
```

Comment:
suggested edit for line 215:
``for resource in Resource.objects.filter(resourceinstance__in=resources):``

Reply:
I've change changed this and am testing.

Reply:
this failed testing so reverted - see comment below